src/lib.rs

```diff
@@ -3828,6 +3838,33 @@ impl Connection {
         self.closed
     }
 
+    /// Returns true if the connection is closing.
+    #[inline]
+    pub fn is_closing(&self) -> bool {
```

Comment:
This function doesn't actually do what the description say it does. It only returns true if the local endpoint set an error, not whether the error is actually sent (so the closing process is actually initiated), or whether an error is _received_.

Since we already have the `is_draining()` method that covers all of the above, we shouldn't add this, as it simply adds confusion and doesn't seem to be particularly useful anyway.

Reply:
I'll take some blame here. In the current code in `recv_single` and `send_single` there's `is_closing` based on has a view of whether the connection has an error or not. So I think I suggested we could make an `is_closing()` method. But I do agree with ghedo that this can be a confusing public API, so apologies for any churn. Since the check is now just a single parameter check it doesn't save much to wrap it in a method.